ui-speedspacechart/src/components/SpeedSpaceChart.tsx

```diff
@@ -54,15 +75,23 @@ const SpeedSpaceChart = ({ width, height, backgroundColor, data }: SpeedSpaceCha
       stops: data.simulation.present.trains[0].base.stops || [],
       electrification: data.simulation.present.trains[0].electrification_ranges || [],
       slopes: data.simulation.present.trains[0].slopes || [],
+      electricalProfiles: data.electricalProfiles,
     };
 
-    if (storeData.speed && storeData.stops && storeData.electrification && storeData.slopes) {
+    if (
+      storeData.speed &&
+      storeData.stops &&
+      storeData.electrification &&
+      storeData.slopes &&
+      storeData.electricalProfiles
```

Comment:
electrical profiles have to be optional, should be tested in another condition

Reply:
You're right. But here the condition will always be true. Do we need to keep it?